### DISCUSSION TITLE
Support for actually-inline svg visuals?

### DISCUSSION THREAD

**Discussion by cemerick** [marp-team/marp#106] [Q&A] [answered]
I know there is an option called `inlineSVG`, but I gather that that is related to the how slides are wrapped to allow for various scaling effects, etc.

Separate from that, I would like to be able to use inline svg content to define slide visuals, e.g.:

```svg
<svg
   xmlns="http://www.w3.org/2000/svg"
   version="1.1"
   width="620.48627"
   height="620.48627">
  <g
     id="g4534"
     transform="matrix(3.7795276,0,0,3.7795276,-63.513319,-330.4338)">
    <circle
       id="circle4528"
       style="opacity:1;fill:#008000;fill-opacity:1;stroke:none;stroke-width:7.30687523;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       r="82.085159"
       cy="169.51244"
       cx="98.889725" />
  </g>
</svg>
```

This is rendered as expected if the markdown document is passed through `pandoc`, but does not show up in marp's generated PDFs (including when `html: true` and `inlineSVG: true` are set, FWIW).

**Comment by yhatt** [comment]
That may be clearly a bug. The transformed HTML has over-contained `<br />`.

```html
<p><svg
   xmlns="http://www.w3.org/2000/svg"
   version="1.1"
   width="620.48627"
   height="620.48627"><br />
<g
     id="g4534"
     transform="matrix(3.7795276,0,0,3.7795276,-63.513319,-330.4338)"><br />
<circle id="circle4528"
       style="opacity:1;fill:#008000;fill-opacity:1;stroke:none;stroke-width:7.30687523;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1"
       r="82.085159"
       cy="169.51244"
       cx="98.889725" /><br />
</g><br />
</svg></p>
```

It works if not contained line breaks within SVG:

```markdown
---
marp: true
---

<svg xmlns="http://www.w3.org/2000/svg" version="1.1" width="620.48627" height="620.48627"><g id="g4534" transform="matrix(3.7795276,0,0,3.7795276,-63.513319,-330.4338)"><circle id="circle4528" style="opacity:1;fill:#008000;fill-opacity:1;stroke:none;stroke-width:7.30687523;stroke-miterlimit:4;stroke-dasharray:none;stroke-opacity:1" r="82.085159" cy="169.51244" cx="98.889725" /></g></svg>
```

![スクリーンショット 2021-05-05 2 27 25](https://user-images.githubusercontent.com/3993388/117044504-7873b980-ad49-11eb-8d7d-668d29ac33a2.png)

**Comment by cemerick** [comment, marked as answer]
I didn't have to remove any linebreaks; just wrapping the `svg` element with a `p` or a `div` yielded the expected output.

So maybe a bug, but thanks for the workaround! :pray:

**Reply by yhatt** [reply]
Yeah, wrapping `<div>` or `<p>` will recognize the whole of content as HTML block. :smile: 

I made a ticket https://github.com/marp-team/marpit/issues/292 to fix the first case. Track it if interested!
